Makefile

```diff
@@ -183,6 +183,12 @@ run-ci-e2e-test:
 run-ci-e2e-upgrade-test:
 	hack/run-ci-e2e-test.sh -t upgrade
 
+# TODO: Consolidate run-ci-e2e-* targets into a single target with arguments.
+#  For example: run-ci-e2e-test $(ARGS). Requires updates in release steps.
+.PHONY: run-ci-e2e-byoh-test $(ARGS)
+run-ci-e2e-byoh-test:
+	hack/run-ci-e2e-test.sh ${ARGS}
```

Comment:
I would rather have this explicit now instead of using $(ARGS). Once you have introduced the new single target with $(ARGS), you can collapse all the existing endpoints. The reason for this is that we will eventually get rid of `run-ci-e2e-byoh-test` and there is little advantage in doing this here.

Reply:
That was my initial implementation too, the main driver for using a parameter here is:
- the windows instance must be provisioned before invoking the test
- the test is invoked with arguments (`$ARGS`) right after  `$WINDOWS_INSTANCES_DATA` is set as env var.
- `$WINDOWS_INSTANCES_DATA` and `$ARGS` must be aligned, given `$ARGS` contains the number of entries in `$WINDOWS_INSTANCES_DATA` 

```
make run-ci-e2e-byoh-test ARGS="-t basic -m 0 -c 1"
```

So, having both definitions in the same place is safer.

Reply:
I don't follow your reasoning here. Why can't you use the `$WINDOWS_INSTANCES_DATA` in the call here then without having to use `$(ARGS)`?

> So, having both definitions in the same place is safer.

I don't understand what you mean by this.

Reply:
By `same place` meant the [commands](https://github.com/jrvaldes/release/blob/5ab5b2e8790f7cd7b7b11cb8e20019e366741d5d/ci-operator/step-registry/windows/e2e/operator/test/byoh/windows-e2e-operator-test-byoh-commands.sh#L43) in the release repo.

You are right, will adjust to only pass `WINDOWS_INSTANCES_DATA` and parse the number of instances [programmatically](https://github.com/openshift/windows-machine-config-operator/pull/858#discussion_r780426489) in the e2e code.

Reply:
That is not what I mean. We should just have a `make run-ci-e2e-byoh-test` in the release repo without any args. This is what we have been doing in the past mainly because we want to reduce the interaction we have with the release repo. With the pattern you are introducing anytime we want to change the way we are calling the test we have to go to the release repo instead of the WMCO repo. IOW, don't introduce `$(ARGS)`.

Reply:
I follow, by
> only pass WINDOWS_INSTANCES_DATA

I meant, pass `WINDOWS_INSTANCES_DATA` as an environment variable. 

The Makefile target has no arguments, and the test can be configured and invoked using the pattern from a release job and/or running locally. For example:

```
export WINDOWS_INSTANCES_DATA

make run-ci-e2e-test
```

So, no need to introduce the proposed `run-ci-e2e-byoh-test` target at all. 

`hack/run-ci-e2e-test.sh` will look after `$WINDOWS_INSTANCES_DATA` and create the ConfigMap with the provided data if any. At this point, syntax validation is deferred to `oc apply ` API.

WMCO e2e test must be smart enough, to process an existing `windows-instances` ConfigMap and adjust accordingly. i.e. do not create BOYH instances on platform-agnostic infrastructure and use the provided ones, if any.